File: rb/features.rb
# DymoApiIntroduction SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module DymoApiIntroductionFeatures
  def self.make_feature(name)
    case name
    when "base"
      DymoApiIntroductionBaseFeature.new
    when "test"
      DymoApiIntroductionTestFeature.new
    else
      DymoApiIntroductionBaseFeature.new
    end
  end
end

File: py/features.py
# DymoApiIntroduction SDK feature factory

from feature.base_feature import DymoApiIntroductionBaseFeature
from feature.test_feature import DymoApiIntroductionTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DymoApiIntroductionBaseFeature(),
        "test": lambda: DymoApiIntroductionTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
